File: Server/src/main/content/global/skill/combat/summoning/pet/IncubatorEgg.java
package content.global.skill.combat.summoning.pet;

import core.game.node.item.Item;

/**
 * The enum Incubator egg.
 */
public enum IncubatorEgg {
    /**
     * The Penguin.
     */
    PENGUIN(new Item(12483), 30, 30, new Item(12481)),
    /**
     * The Raven.
     */
    RAVEN(new Item(11964), 50, 30, new Item(12484)),
    /**
     * The Saradomin owl.
     */
    SARADOMIN_OWL(new Item(5077), 70, 60, new Item(12503)),
    /**
     * The Zamorak hawk.
     */
    ZAMORAK_HAWK(new Item(5076), 70, 60, new Item(12506)),
    /**
     * The Guthix raptor.
     */
    GUTHIX_RAPTOR(new Item(5078), 70, 60, new Item(12509)),
    /**
     * The Vulture.
     */
    VULTURE(new Item(11965), 85, 60, new Item(12498)),
    /**
     * The Chameleon.
     */
    CHAMELEON(new Item(12494), 90, 60, new Item(12492)),
    /**
     * The Red dragon.
     */
    RED_DRAGON(new Item(12477), 99, 60, new Item(12469)),
    /**
     * The Black dragon.
     */
    BLACK_DRAGON(new Item(12480), 99, 60, new Item(12475)),
    /**
     * The Blue dragon.
     */
    BLUE_DRAGON(new Item(12478), 99, 60, new Item(12471)),
    /**
     * The Green dragon.
     */
    GREEN_DRAGON(new Item(12479), 99, 60, new Item(12473));

    private final Item egg;

    private final int level;

    private final int inucbationTime;

    private final Item product;

    IncubatorEgg(Item egg, int level, int inucbationTime, Item product) {
        this.egg = egg;
        this.level = level;
        this.inucbationTime = inucbationTime;
        this.product = product;
    }

    /**
     * For item incubator egg.
     *
     * @param item the item
     * @return the incubator egg
     */
    public static IncubatorEgg forItem(Item item) {
        for (IncubatorEgg e : values()) {
            if (e.getEgg().getId() == item.getId()) {
                return e;
            }
        }
        return null;
    }

    /**
     * Gets egg.
     *
     * @return the egg
     */
    public Item getEgg() {
        return egg;
    }

    /**
     * Gets level.
     *
     * @return the level
     */
    public int getLevel() {
        return level;
    }

    /**
     * Gets incubation time.
     *
     * @return the incubation time
     */
    public int getIncubationTime() {
        return inucbationTime;
    }

    /**
     * Gets product.
     *
     * @return the product
     */
    public Item getProduct() {
        return product;
    }

}

File: Emulator/src/main/content/global/skill/combat/summoning/pet/IncubatorEgg.kt
package content.global.skill.combat.summoning.pet

import cfg.consts.Items
import core.game.node.item.Item

/**
 * Incubator egg
 *
 * @param egg Represents the item that is the egg.
 * @param level Indicates the level required for incubation.
 * @param incubationTime Specifies the time needed for the egg to incubate.
 * @param product Represents the item produced after incubation.
 * @constructor Incubator egg
 */
enum class IncubatorEgg(
    val egg: Item, // The egg item being incubated
    val level: Int, // The level required for successful incubation
    val incubationTime: Int, // The duration of the incubation process
    val product: Item // The resulting item after incubation
) {
    /**
     * Penguin
     *
     * @constructor Penguin
     */
    PENGUIN(
        egg = Item(Items.PENGUIN_EGG_12483),
        level = 30,
        incubationTime = 30,
        product = Item(Items.BABY_PENGUIN_12481)
    ),

    /**
     * Raven
     *
     * @constructor Raven
     */
    RAVEN(
        egg = Item(Items.RAVEN_EGG_11964),
        level = 50,
        incubationTime = 30,
        product = Item(Items.RAVEN_CHICK_12484)
    ),

    /**
     * Saradomin Owl
     *
     * @constructor Saradomin Owl
     */
    SARADOMIN_OWL(
        egg = Item(Items.BIRDS_EGG_5077),
        level = 70,
        incubationTime = 60,
        product = Item(Items.SARADOMIN_CHICK_12503)
    ),

    /**
     * Zamorak Hawk
     *
     * @constructor Zamorak Hawk
     */
    ZAMORAK_HAWK(
        egg = Item(Items.BIRDS_EGG_5076),
        level = 70,
        incubationTime = 60,
        product = Item(Items.ZAMORAK_CHICK_12506)
    ),

    /**
     * Guthix Raptor
     *
     * @constructor Guthix Raptor
     */
    GUTHIX_RAPTOR(
        egg = Item(Items.BIRDS_EGG_5078),
        level = 70,
        incubationTime = 60,
        product = Item(Items.GUTHIX_CHICK_12509)
    ),

    /**
     * Vulture
     *
     * @constructor Vulture
     */
    VULTURE(
        egg = Item(Items.VULTURE_EGG_11965),
        level = 85,
        incubationTime = 60,
        product = Item(Items.VULTURE_CHICK_12498)
    ),

    /**
     * Chameleon
     *
     * @constructor Chameleon
     */
    CHAMELEON(
        egg = Item(Items.CHAMELEON_EGG_12494),
        level = 90,
        incubationTime = 60,
        product = Item(Items.BABY_CHAMELEON_12492)
    ),

    /**
     * Red Dragon
     *
     * @constructor Red Dragon
     */
    RED_DRAGON(
        egg = Item(Items.RED_DRAGON_EGG_12477),
        level = 99,
        incubationTime = 60,
        product = Item(Items.HATCHLING_DRAGON_12469)
    ),

    /**
     * Black Dragon
     *
     * @constructor Black Dragon
     */
    BLACK_DRAGON(
        egg = Item(Items.BLACK_DRAGON_EGG_12480),
        level = 99,
        incubationTime = 60,
        product = Item(Items.HATCHLING_DRAGON_12475)
    ),

    /**
     * Blue Dragon
     *
     * @constructor Blue Dragon
     */
    BLUE_DRAGON(
        egg = Item(Items.BLUE_DRAGON_EGG_12478),
        level = 99,
        incubationTime = 60,
        product = Item(Items.HATCHLING_DRAGON_12471)
    ),

    /**
     * Green Dragon
     *
     * @constructor Green Dragon
     */
    GREEN_DRAGON(
        egg = Item(Items.GREEN_DRAGON_EGG_12479),
        level = 99,
        incubationTime = 60,
        product = Item(Items.HATCHLING_DRAGON_12473)
    );

    companion object {

        fun forItem(item: Item): IncubatorEgg? {
            for (e in values()) {
                if (e.egg.id == item.id) {
                    return e
                }
            }
            return null
        }
    }
}